src/pages/index.js

```diff
@@ -23,88 +37,233 @@ const profileDescriptionInput = document.querySelector(
 const section = new Section(
   {
-    items: initialCards,
+    items: [],
     renderer: (cardData) => {
+      //
       const cardElement = createCard(cardData);
+      //
       section.addItem(cardElement);
     },
   },
   ".cards"
 );
 
-section.renderItems();
+console.log("Section instance:", section);
 
-// Modal Event Listeners
-editProfileModal.setEventListeners();
-addCardModal.setEventListeners();
-imagePreviewModal.setEventListeners();
+// Fetch and Render Initial Data//
+api
+  .getAllData()
+  .then(({ user, cards }) => {
+    console.log("Initial user data:", user);
+    console.log("Initial cards data:", cards);
+    section.renderItems(cards);
+    userInfo.setUserInfo(user);
+  })
+  .catch((error) => console.error("Error loading data", error));
+
+const editAvatarModal = new PopupWithForm({
+  popupSelector: "#edit-avatar-modal",
+  handleFormSubmit: (data) => {
+    const userAvatar = {
+      avatar: data.avatar,
+    };
+    api
+      .updateUserAvatar(userAvatar)
+      .then((updatedUserInfo) => {
+        userInfo.setUserInfo(updatedUserInfo);
+        editAvatarModal.close();
+      })
+      .catch((error) => console.error("Error updating user avatar:", error));
+  },
+});
+
+const editProfileModal = new PopupWithForm({
+  popupSelector: "#edit-profile-modal",
+  handleFormSubmit: (data) => {
+    const userData = {
+      name: data.name,
+      about: data.about,
+    };
+    editProfileModal.renderLoading(true);
+    api
+      .updateUserInfo(userData)
+      .then((updatedUserInfo) => {
+        userInfo.setUserInfo(updatedUserInfo);
+        editProfileModal.close();
+      })
+      .catch((error) => console.error("Error updating user info:", error))
+      .finally(() => {
+        editProfileModal.renderLoading(false);
+      });
+  },
+});
+
+const addCardModal = new PopupWithForm({
+  popupSelector: "#add-card-modal",
+  handleFormSubmit: (data) => {
+    const cardData = {
+      name: data.title,
+      link: data.url,
+    };
+
+    addCardModal.renderLoading(true);
+    api
+      .addCard(cardData)
+      .then((newCard) => {
+        if (newCard && newCard.name && newCard.link) {
+          const cardElement = createCard(
+            //   {
+            //   name: newCard.name,
+            //   link: newCard.link,
+            //   _id: newCard._id,
+            // }
+            newCard
+          );
+          section.addItem(cardElement);
+          cardForm.reset();
+          toggleAddButtonState();
+          addCardModal.close();
+        } else {
+          console.error("Invalid card data:", newCard);
+        }
+      })
+      .catch((error) => console.error("Error adding new card:", error))
+      .finally(() => {
+        addCardModal.renderLoading(false);
+      });
+  },
+});
+
+const deleteCardModal = new PopupWithConfirm({
+  popupSelector: "#delete-card-modal",
+  handleConfirm: () => {
+    if (currentCard) {
+      handleDeleteCard(currentCard);
+      deleteCardModal.close();
```

Comment:
You need to close modals only in blocks `then` after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well